theano/tensor/nnet/nnet.py

```diff
@@ -384,6 +384,7 @@ def c_code(self, node, name, inp, out, sub):
             }
         }
 
+        #pragma omp parallel for schedule(static, 32)
```

Comment:
Hi nouiz,
Thanks for your testing.

**1. Regarding to conditional openmp for different shape: I agree with you.**
Actually, I want to do it before. But this will bring more code change. However, if you think it's OK. I am OK. I am working on it, will send out the code change later.
Just for your reference, below is my size samples which are driven by RNN/LSTM workload(I'll choose 128, 256000 as the condition to do code split. Surely, I have tested it, it works fine):
`Softmax: 544, 30000, 
Softmax: 480, 30000, 
Softmax: 448, 30000, 
Softmax: 384, 30000, 
Softmax: 320, 30000, 
Softmax: 140, 30000, 
Softmax: 2176, 30000,
Softmax: 1664, 30000,
Softmax: 1536, 30000,
Softmax: 1344, 30000,
Softmax: 1248, 30000,
Softmax: 1184, 30000,
Softmax: 1088, 30000,
Softmax: 1024, 30000,
Softmax: 960, 30000, 
Softmax: 864, 30000, 
Softmax: 800, 30000, 
Softmax: 736, 30000, 
Softmax: 672, 30000, 
Softmax: 608, 30000, 
Softmax: 576, 30000, 
Softmax: 512, 30000, 
Softmax: 416, 30000, 
Softmax: 352, 30000, 
Softmax: 225, 30000, 
Softmax: 2080, 30000,
Softmax: 1600, 30000,
Softmax: 1472, 30000,
Softmax: 1344, 30000,
Softmax: 1280, 30000,
Softmax: 1216, 30000,
Softmax: 1152, 30000,
Softmax: 1088, 30000,
Softmax: 992, 30000, 
Softmax: 960, 30000, 
Softmax: 896, 30000, 
Softmax: 800, 30000, 
Softmax: 736, 30000, 
Softmax: 672, 30000, 
Softmax: 608, 30000, 
Softmax: 544, 30000, 
Softmax: 448, 30000, 
Softmax: 384, 30000, 
Softmax: 42, 30000,
Softmax: 1920, 30000,
Softmax: 1536, 30000,
Softmax: 1376, 30000,
Softmax: 1248, 30000,
Softmax: 1120, 30000,
Softmax: 992, 30000, 
Softmax: 928, 30000, 
Softmax: 800, 30000, 
Softmax: 704, 30000, 
Softmax: 608, 30000, 
Softmax: 512, 30000, 
Softmax: 448, 30000, 
Softmax: 320, 30000, 
Softmax: 48, 30000,`

**2. Regarding to your second comment about SoftmaxWithBias:**
I have consider this before. Actually, it includes SoftmaxWithBias, LogSoftmax, etc. However, there is not the related famous workload(e.g.: Alexnet, Googlenet, dl4mt etc.) to direct our optimization work. My opinion is: don't do it now :). When there is related workload to drive it, we do it. What's your opinion?


BTW, several months before, our team focus on theano CNN optimization. At that time, softmax is not the bottleneck, because there is only about 1000 kinds  classifications result for CNN(e.g.: Alexnet and Googlent). When we try to do theano RNN/LSTM optimization, we found that softmax cost so much CPU time. Because for RNN/LSTM, there are so many kinds classifications.